tests/std/tests/P0919R3_heterogeneous_unordered_lookup/test.cpp

```diff
@@ -22,19 +22,24 @@ using namespace std;
 
 bool g_prohibit_allocations = false;
 
-struct prohibit_allocations {
-    prohibit_allocations() {
-        g_prohibit_allocations = true;
+class prohibit_allocations {
+public:
+    explicit prohibit_allocations(const bool value) : prev(g_prohibit_allocations) {
+        g_prohibit_allocations = value;
     }
 
     prohibit_allocations(const prohibit_allocations&) = delete;
     prohibit_allocations& operator=(const prohibit_allocations&) = delete;
 
     ~prohibit_allocations() {
-        g_prohibit_allocations = false;
+        g_prohibit_allocations = prev;
     }
+
+private:
+    const bool prev;

```

Comment:
This is named the same as `std::prev` in `<iterator>` (which this test is including). Although there's no ambiguity as far as the compiler is concerned, I always recommend avoiding shadowing. Can we rename this to `m_prev` or `previous` or something?